libgramtools/src/search/search.cpp

```diff
@@ -308,8 +308,12 @@ SA_Interval gram::get_allele_marker_sa_interval(const Marker &site_marker_char,
     const auto start_sa_index = prg_info.fm_index.C[alphabet_rank];
 
     const auto next_boundary_marker = allele_marker_char + 1;
-    const auto max_char_in_alphabet = prg_info.fm_index.sigma - 1;
-    const bool next_boundary_marker_valid = next_boundary_marker <= max_char_in_alphabet;
+    // sigma: contains the size (=number of unique symbols) of the alphabet
+    const auto max_alphabet_char = prg_info.fm_index.comp2char[prg_info.fm_index.sigma - 1];
```

Comment:
Nice fix. This no longer assumes that all integers between 1 and sigma are actually being used in the prg (ie prg alphabet can be **discontinuous**.) 

I think there are other places in gramtools where prgs encoded using a discontinuous alphabet would cause trouble. One example is line 310:

 `const auto next_boundary_marker = allele_marker_char + 1; `

If the prg uses for example only 5 and 23 as site markers, then from `allele_marker_char` of 6, we would assign `next_boundary_marker` to 7, which is non-existent.

I will write these similar types of problems where spotted in `TODOs` in `add_doc` branch.